functions/battery/fn_updateBatteryLevel.sqf

```diff
@@ -0,0 +1,162 @@
+#define LAST_TRANSMISSION_RADIO_ID "COLSOG_lastTransmissionRadioId"
+
+if !(hasInterface) exitWith {};
+
+_displayBatteryLevelPrc77 = [
+	"COLSOG_battery",
+	"PRC77 - Show battery level",
+	"\a3\Modules_F_Curator\Data\iconLightning_ca.paa",
+	{
+		["ACRE_PRC77", player] call COLSOG_fnc_displayBatteryLevel;
+	},
+	{
+        [player, "ACRE_PRC77"] call acre_api_fnc_hasKindOfRadio
+	}
+] call ace_interact_menu_fnc_createAction;
+
+["Man", 1, ["ACE_SelfActions", "ACE_Equipment"], _displayBatteryLevelPrc77, true] call ace_interact_menu_fnc_addActionToClass;
+
+_displayBatteryLevelPrc343 = [
+	"COLSOG_battery",
+	"PRC343 - Show battery level",
+	"\a3\Modules_F_Curator\Data\iconLightning_ca.paa",
+	{
+		["ACRE_PRC343", player] call COLSOG_fnc_displayBatteryLevel;
+	},
+	{
+        [player, "ACRE_PRC343"] call acre_api_fnc_hasKindOfRadio
+	}
+] call ace_interact_menu_fnc_createAction;
+
+["Man", 1, ["ACE_SelfActions", "ACE_Equipment"], _displayBatteryLevelPrc343, true] call ace_interact_menu_fnc_addActionToClass;
+
+
+_increaseBatteryLevelPrc77 = [
+	"COLSOG_battery",
+	"PRC77 - Add new battery",
+	"x\zen\addons\context_actions\ui\add_ca.paa",
+	{
+		["ACRE_PRC77", player] call COLSOG_fnc_increaseBatteryLevel;
+	},
+	{
+	    private _result = false;
+
+	    private _hasPowerItem = false;
+        {
+            if ([player, _x] call BIS_fnc_hasItem) then
+            {
+                _hasPowerItem = true;
+            };
+        } forEach colsog_battery_powerItems;
+
+        private _hasRadioItem = ([player, "ACRE_PRC77"] call acre_api_fnc_hasKindOfRadio);
+
+        _result = (_hasPowerItem AND _hasRadioItem);
+        _result
+	}
+] call ace_interact_menu_fnc_createAction;
+
+["Man", 1, ["ACE_SelfActions", "ACE_Equipment"], _increaseBatteryLevelPrc77, true] call ace_interact_menu_fnc_addActionToClass;
+
+
+_increaseBatteryLevelPrc343 = [
+	"COLSOG_battery",
+	"PRC343 - Add new battery",
+	"x\zen\addons\context_actions\ui\add_ca.paa",
+	{
+		["ACRE_PRC343", player] call COLSOG_fnc_increaseBatteryLevel;
+	},
+	{
+	    private _result = false;
+
+	    private _hasPowerItem = false;
+        {
+            if ([player, _x] call BIS_fnc_hasItem) then
+            {
+                _hasPowerItem = true;
+            };
+        } forEach colsog_battery_powerItems;
+
+        private _hasRadioItem = ([player, "ACRE_PRC343"] call acre_api_fnc_hasKindOfRadio);
+
+        _result = (_hasPowerItem AND _hasRadioItem);
+        _result
+	}
+] call ace_interact_menu_fnc_createAction;
+
+["Man", 1, ["ACE_SelfActions", "ACE_Equipment"], _increaseBatteryLevelPrc343, true] call ace_interact_menu_fnc_addActionToClass;
+
+
+// Wait for first broadcast in order to initialize battery on all radio of current player.
+[
+    {[player] call acre_api_fnc_isBroadcasting;},
+    {execVM "functions\battery\fn_initializeBatteryLevel.sqf";},
+    [player]
+]
```

Comment:
After refactoring, I feel like this is not applying properly at initial radio call. 

When initializing radio battery using "display level" or "add new battery" equipment function instead of directly talking over radio (which should initialize them, and was working before refactoring) I have the below error.

![20240824234523_1](https://github.com/user-attachments/assets/35ac7f37-959a-4a8c-afa7-021867dd31d8)

But nonetheless, everything seems to work nicely in local multiplayer. Regardless of this error  :/